File: backend/test_post.py
#!/usr/bin/env python3
"""Create a test post and verify it writes to the database."""
import os
import sys

from dotenv import load_dotenv
BASE_DIR = os.path.dirname(os.path.abspath(__file__))
load_dotenv(os.path.join(BASE_DIR, ".env"))

# Add backend to path
sys.path.insert(0, BASE_DIR)

from app import _create_post

def main():
    if not os.getenv("DATABASE_URL") and not os.getenv("PGHOST"):
        print("Error: Set DATABASE_URL or PGHOST/PGPASSWORD in .env")
        sys.exit(1)

    post = _create_post(
        title="Test Post from Script",
        excerpt="A test post to verify database writes.",
        content="This post was created by test_post.py to verify the blog app writes to the database correctly.",
    )

    if post:
        print("Success! Test post created:")
        print(f"  ID: {post['id']}")
        print(f"  Title: {post['title']}")
        print(f"  Date: {post['date']}")
        print("\nPost written to database.")
    else:
        print("Error: Failed to create post. Check DATABASE_URL and database connection.")
        sys.exit(1)

if __name__ == "__main__":
    main()

File: scripts/test-post.js
/**
 * Create a test post and verify it writes to the database.
 * Run: node --experimental-vm-modules scripts/test-post.js
 * Requires HASURA_GRAPHQL_ENDPOINT and HASURA_ADMIN_SECRET.
 */
import { config } from "dotenv";
import path from "path";
import { fileURLToPath } from "url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));

// Load .env.local from project root
config({ path: path.join(__dirname, "..", ".env.local") });
config({ path: path.join(__dirname, "..", ".env") });

async function main() {
  const { addEntry, getEntries } = await import("../src/lib/blogEntries.js");

  if (!process.env.HASURA_GRAPHQL_ENDPOINT || !process.env.HASURA_ADMIN_SECRET) {
    console.error("Error: Set HASURA_GRAPHQL_ENDPOINT and HASURA_ADMIN_SECRET in .env.local");
    process.exit(1);
  }

  const testPost = {
    title: "Test Post from Script",
    excerpt: "A test post to verify database writes.",
    content:
      "This post was created by scripts/test-post.js to verify the blog app writes to the database correctly.",
  };

  try {
    const post = await addEntry(testPost);
    console.log("Success! Test post created:");
    console.log("  ID:", post.id);
    console.log("  Title:", post.title);
    console.log("  Date:", post.date);
    console.log("\nVerifying in database...");

    const posts = await getEntries();
    const found = posts.find((p) => p.id === post.id);
    if (found) {
      console.log("Confirmed: Post found in database.");
    } else {
      console.log("Warning: Post created but not found in getEntries.");
    }
  } catch (err) {
    console.error("Error:", err.message);
    process.exit(1);
  }
}

main();
